content/docs/io/overview.md

```diff
@@ -6,7 +6,119 @@ menu:
     parent: io
 ---
 
-This page has not been worked on yet. If you'd like to contribute visit the [doc-push]
-repo.
+The Rust standard library provides support for networking and I/O, such
+as TCP connections, UDP sockets, reading from and writing to files, etc.
+However, those operations are all synchronous, or _blocking_, meaning
+that when you call them, the current thread may stop executing and go to
+sleep until it is unblocked. For example, `read` will block until there
+is data to read. In the world of futures, that behavior is unfortunate,
+since we would like to continue executing other futures we may have
+while waiting for the I/O to complete.
 
-[doc-push]: https://github.com/tokio-rs/doc-push
+To enable this, Tokio provides _non-blocking_ versions of many standard
+library I/O resources, such as [file operations] and [TCP], [UDP], and
+[Unix] sockets. They return futures for long-running operations (like
+accepting a new TCP connection), and implement non-blocking variants of
+`std::io::Read` and `std::io::Write` called `AsyncRead` and
+`AsyncWrite`.
+
+Non-blocking reads and writes do not block if, for example, there is no
+more data available. Instead, they return immediately with a
+`WouldBlock` error, along with a promise (like `Future::poll`) that they
```

Comment:
I find the use of the word "promise" here to be potentially problematic. I only understand what you mean because I know the implementation. These calls don't _return_ a promise (which seems to be implied here).

Reply:
Yeah, we probably should not use `promise` here given that it is overloaded.